src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/Options.cs

```diff
@@ -24,64 +26,82 @@ public enum SettingFields : uint
         /// <summary>
-        /// Specific label of the key.
+        /// Keys that will be used to filter.
         /// </summary>
-        public string Label { get; set; } = null;
-
-        public ETagFilter ETag { get; set; }
-
+        /// <remarks>See the documentation for this client library for details on the format of filter expressions</remarks>
+        public IList<string> Keys { get; set; }
         /// <summary>
-        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// Labels that will be used to filter.
         /// </summary>
-        public DateTimeOffset? Revision { get; set; }
-
+        /// <remarks>See the documentation for this client library for details on the format of filter expressions</remarks>
+        public IList<string> Labels { get; set; }
         /// <summary>
         /// IKeyValue fields that will be retrieved.
         /// </summary>
         public SettingFields Fields { get; set; } = SettingFields.All;
-        
-        public static implicit operator RequestOptions(string label) => new RequestOptions() { Label = label };
+        /// <summary>
+        /// If set, then key values will be retrieved exactly as they existed at the provided time.
+        /// </summary>
+        public DateTimeOffset? AsOf { get; set; }
 
-        #region nobody wants to see these
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public override bool Equals(object obj) => base.Equals(obj);
+        public SettingSelector() : this(Any, Any) { }
 
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public override int GetHashCode() => base.GetHashCode();
+        public SettingSelector(string key, string label = default)
+        {
+            Keys = new List<string> { key };
+            Labels = new List<string>();
+            if (label != null) Labels.Add(label);
+        }
 
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        // TODO ()
-        public override string ToString() => base.ToString();
-        #endregion
-    }
+        internal string BatchLink { get; set; }
 
-    public class BatchRequestOptions : RequestOptions
-    {
-        /// <summary>
-        /// Keys that will be used to filter.
-        /// </summary>
-        /// <remarks>See the documentation for this SDK for details on the format of filter expressions</remarks>
-        public string Key { get; set; } = "*";
+        internal SettingSelector CloneWithBatchLink(string batchLink)
+        {
+            return new SettingSelector()
+            {
+                Keys = new List<string>(Keys),
+                Labels = new List<string>(Labels),
+                Fields = Fields,
+                AsOf = AsOf,
+                BatchLink = batchLink
+            };
+        }
 
-        public string BatchLink { get; set; }
+        #region nobody wants to see these
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
This attribute should be applied to `Equals(SettingSelector other)` only to `override bool Equals(object obj)`